src/Sentry/Internal/MainExceptionProcessor.cs

```diff
@@ -49,7 +56,30 @@ public void Process(Exception exception, SentryEvent sentryEvent)
 
                 foreach (var key in sentryException.Data.Keys)
                 {
-                    sentryEvent.SetExtra($"Exception[{i}][{key}]", sentryException.Data[key]);
+                    if (key.StartsWith("sentry", StringComparison.OrdinalIgnoreCase))
+                    {
+                        if (key.StartsWith(ExceptionDataTagKey, StringComparison.OrdinalIgnoreCase))
```

Comment:
Better then to iterate the map instead of only the keys though. You'd get back a KeyValuePair<T,V>.